hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/ComputeHash.java

```diff
@@ -0,0 +1,954 @@
+                    }
+
+                    if (computeContent) {
+                        RelativePath mtPath = relativize(metric.getMetricTypePath(), SegmentType.mt, childContext);
+                        appendContent(mtPath.toString(), childContext);
+                        appendContent(Objects.toString(metric.getCollectionInterval(), ""), childContext);
+                        appendCommonContent(metric, childContext);
+                    }
+
+                    if (computeSync) {
+                        appendCommonSync(metric, childContext);
+                    }
+                });
+            }
+
+            @Override
+            public IntermediateHashResult visitResource(Resource.Blueprint resource,
+                                                        IntermediateHashContext context) {
+                return wrap(resource, context, (childContext) -> {
+                    if (computeIdentity) {
+                        appendEntityIdentity(structure.getConfiguration(context.root, resource), childContext);
+                        appendEntityIdentity(structure.getConnectionConfiguration(context.root, resource),
+                                childContext);
+                        structure.getResources(context.root, resource)
+                                .forEach(b -> appendEntityIdentity(b, childContext));
+                        structure.getResourceMetrics(context.root, resource)
+                                .forEach(b -> appendEntityIdentity(b, childContext));
+                        appendIdentity(resource.getId(), childContext);
+                    }
+
+                    if (computeContent) {
+                        RelativePath rtPath = relativize(resource.getResourceTypePath(), SegmentType.rt, childContext);
+                        appendContent(rtPath.toString(), childContext);
+                        appendCommonContent(resource, childContext);
+                    }
+
+                    if (computeSync) {
+                        appendCommonSync(resource, childContext);
+                    }
+                });
+            }
+
+            @Override
+            public IntermediateHashResult visitTenant(Tenant.Blueprint tenant, IntermediateHashContext context) {
+                return wrap(tenant, context, childContext -> {
+                    if (computeContent) {
+                        appendCommonContent(tenant, childContext);
+                    }
+                });
+            }
+
+            @Override
+            public IntermediateHashResult visitEnvironment(Environment.Blueprint environment,
+                                                           IntermediateHashContext context) {
+                return wrap(environment, context, childContext -> {
+                    if (computeContent) {
+                        appendCommonContent(environment, childContext);
+                    }
+                });
+            }
```

Comment:
You're right and it is intentional :smile: The content hash is not hierarchical. Actually the content hash computation should not call itself for it's descendants (which I think you really meant).